### PR TITLE
feat(deploy): print edge function logs url

### PR DESCRIPTION
**Pull request by serhalp**
### Summary

The `netlify deploy` command prints the build logs URL and the function logs URL:

Screenshot 2024-10-02 at 10 59 27

It's inconvenient and inconsistent that this doesn't also include a link to edge function logs. So this PR adds it.

#### bee888c88fd0a7c28aa4b3c5204ce57f2ed669eb: fix issues uncovered by above

- `name` in the JSON output was coming from a field that did not exist, so it was always undefined (and `undefined` values are omitted when JSON stringifying), so I just removed it, as it's effectively never truly existed
  - I'm not sure what this was supposed to be anyway. There's already a `site_name` field which is working as intended.
- `site_id` never worked (or at least hasn't for some time) as it was reading from a field that didn't exist. I fixed this one.

#### 8bee1a9290c7ae481f1985d7ee0acf08df3eea70: add edge function logs URL to pretty output

Example:
Screenshot 2024-10-02 at 11 00 49

#### aed9afed559761f4f6aee5c93fc9dfd2bd0e5889: add `function_logs` and `edge_function_logs` to `--json` output

It wasn't consistent that these weren't included in the JSON output. These also could be useful to users. But mostly it allowed me to more easily add test coverage (none of the printed URLs — neither pretty nor json — had test coverage).

Example before:

```json
{
  "site_name": "hydrogen-remix-template",
  "deploy_id": "66fd5eacf914aa188272ebe8",
  "deploy_url": "https://66fd5eacf914aa188272ebe8--hydrogen-remix-template.netlify.app",
  "logs": "https://app.netlify.com/sites/hydrogen-remix-template/deploys/66fd5eacf914aa188272ebe8"
}
```

Example after:

```json
{
  "site_id": "da7ae002-b3be-4dd5-a294-d44321e2459f",
  "site_name": "hydrogen-remix-template",
  "deploy_id": "66fd5ef32443dc1423ce705b",
  "deploy_url": "https://66fd5ef32443dc1423ce705b--hydrogen-remix-template.netlify.app",
  "logs": "https://app.netlify.com/sites/hydrogen-remix-template/deploys/66fd5ef32443dc1423ce705b",
  "function_logs": "https://app.netlify.com/sites/hydrogen-remix-template/logs/functions?scope=deploy:66fd5ef32443dc1423ce705b",
  "edge_function_logs": "https://app.netlify.com/sites/hydrogen-remix-template/logs/edge-functions?scope=deploy:66fd5ef32443dc1423ce705b"
}
```

### Future improvements

💡 We should only include the function logs URL when the deploy included >0 functions and only include the edge function logs URL when the deploy included >0 edge functions.

### CODE REVIEW

File: src/commands/deploy/deploy.ts [hunk 1]
```diff
@@ -656,16 +669,16 @@ const printResults = ({ deployToProduction, isIntegrationDeploy, json, results,
...
-      name: results.name,
-      site_id: results.site_id,
+    const jsonData: JsonData = {
+      site_id: results.siteId,
       site_name: results.siteName,
       deploy_id: results.deployId,
       deploy_url: results.deployUrl,
       logs: results.logsUrl,
+      function_logs: results.functionLogsUrl,
+      edge_function_logs: results.edgeFunctionLogsUrl,
```
Comment (serhalp) [inline src/commands/deploy/deploy.ts:679]: I would rather have suffixed these with `_url` but since `logs` was already named `logs` I erred toward internal consistency 😞.

File: tests/integration/commands/deploy/deploy.test.ts [hunk 2]
```diff
@@ -263,6 +274,57 @@ describe.skipIf(process.env.NETLIFY_TEST_DISABLE_LIVE === 'true').concurrent('co
...
+  test('should print production URLs for build logs, function logs, and edge function logs when --prod is passed', async (t) => {
+    await withSiteBuilder(t, async (builder) => {
+      const content = '<h1>Why Next.js is perfect, a novella</h1>'
+      builder.withContentFile({
+        path: 'public/index.html',
+        content,
+      })
+      await builder.build()
+
+      const deploy = await callCli(['deploy', '--json', '--dir', 'public', '--prod'], {
```
Comment (serhalp) [inline tests/integration/commands/deploy/deploy.test.ts:313]: fun fact: this is the very first test coverage of `ntl deploy --prod` 😬😬😬

File: src/commands/deploy/deploy.ts [hunk 3]
```diff
@@ -624,30 +627,40 @@ const bundleEdgeFunctions = async (options, command: BaseCommand) => {
...
+  deployToProduction: boolean
+  isIntegrationDeploy: boolean
+  json: boolean
+  results: Awaited<ReturnType<typeof prepAndRunDeploy>>
+  runBuildCommand: boolean
+}): void => {
+  const msgData: Record<string, string> = {
     'Build logs': results.logsUrl,
     'Function logs': results.functionLogsUrl,
+    'Edge Function Logs': results.edgeFunctionLogsUrl,
```
Comment (eduardoboucas) [inline src/commands/deploy/deploy.ts:657]: nit: Edge Functions is only capitalised when referring to the product (plural), not a single function (singular).

```suggestion
    'Edge function Logs': results.edgeFunctionLogsUrl,
```

File: src/commands/deploy/deploy.ts [hunk 4]
```diff
@@ -656,16 +669,16 @@ const printResults = ({ deployToProduction, isIntegrationDeploy, json, results,
 
   // Json response for piping commands
   if (json) {
-    const jsonData = {
-      name: results.name,
```
Comment (eduardoboucas) [inline src/commands/deploy/deploy.ts:660]: Did you mean to remove `name`?

Comment (eduardoboucas) [reply src/commands/deploy/deploy.ts:660]: Never mind, just saw the PR description.

**Review by eduardoboucas — APPROVED** [review]
